drivers/sftp/onitu_sftp/tests/driver.py

```
@@ -0,0 +1,108 @@
+import os
+import hashlib
+import paramiko
+
+from tests.utils.testdriver import TestDriver
+
+
+class Driver(TestDriver):
+    SPEED_BUMP = 1
+
+    def __init__(self, *args, **options):
+
+        # Clean the root
+        root = options['root']
+        if root.endswith('/'):
+            root = root[:-1]
+
+        hostname = options['hostname']
+        username = options['username']
+        password = options['password']
+        port = options['port']
+        private_key_passphrase = options['private_key_passphrase']
+        private_key_path = options['private_key_path']
+
+        private_key_file = os.path.expanduser(private_key_path)
+        private_key_error = None
+        try:
+            private_key = paramiko.RSAKey.from_private_key_file(
+                private_key_file, password=private_key_passphrase)
+        except IOError as e:
+            private_key_error = e
+
+        transport = paramiko.Transport((hostname, port))
+        try:
+            if password != '':
+                transport.connect(username=username, password=password)
+            else:
+                if private_key_error is not None:
+                    print "SFTP driver connection failed : {}".format(
```

Not Python 3 compatible. Also I guess you can raise an exception rather than printing it.